<a href="https://colab.research.google.com/github/Byng-hee/YOLOv8/blob/main/SecureRange.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
# !wget -O Gun_Data https://app.roboflow.com/ds/0XThtNpas9?key=xc0LjX8eel

--2024-03-31 06:35:08--  https://app.roboflow.com/ds/0XThtNpas9?key=xc0LjX8eel
Resolving app.roboflow.com (app.roboflow.com)... 151.101.1.195, 151.101.65.195
Connecting to app.roboflow.com (app.roboflow.com)|151.101.1.195|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://storage.googleapis.com/roboflow-platform-regional-exports/2GyRAI76lVanezelsFHMGzDtdQy2/Bz9Kt5Y936yaotZbf6n9/1/yolov8.zip?X-Goog-Algorithm=GOOG4-RSA-SHA256&X-Goog-Credential=481589474394-compute%40developer.gserviceaccount.com%2F20240331%2Fauto%2Fstorage%2Fgoog4_request&X-Goog-Date=20240331T063508Z&X-Goog-Expires=901&X-Goog-SignedHeaders=host&X-Goog-Signature=cd989193a52dae2f499e59b5a8e90d12c147dc6b6f29e356508797ef20940849b14d1199fe9812a591f06dd4a6b17cd183ae0d25fdb4a552f594d01fbf6c21e3c9986d8b92c28fe56eaf5266a72c29c6bf3543b2940cfeb1a4b24f11de1520e7f039062dd15d29473799677fd0325d2145cc91ddca8228f67ca1c1f359d965ce1a5840d1fd044a537c1015aabf0a3d36c811cdd9098a4e5263d580511c65e8cc48e9cd3ed0

In [ ]:
from google.colab import drive
import os

drive.mount('/content/drive/')

Mounted at /content/drive/


In [ ]:
# Install the ultralytics package from GitHub
!pip install git+https://github.com/ultralytics/ultralytics.git@main

NotImplementedError: A UTF-8 locale is required. Got ANSI_X3.4-1968

In [ ]:
# !wget -O Gun_Data.zip https://app.roboflow.com/ds/0XThtNpas9?key=xc0LjX8eel
!wget -O Firearms.zip https://app.roboflow.com/ds/D8RqX1rbhn?key=Ai5dmHJsU4

--2024-04-04 03:03:54--  https://app.roboflow.com/ds/D8RqX1rbhn?key=Ai5dmHJsU4
Resolving app.roboflow.com (app.roboflow.com)... 151.101.1.195, 151.101.65.195
Connecting to app.roboflow.com (app.roboflow.com)|151.101.1.195|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://storage.googleapis.com/roboflow-platform-regional-exports/2GyRAI76lVanezelsFHMGzDtdQy2/IGF2I44RsU5Rym6kiTX3/1/yolov8.zip?X-Goog-Algorithm=GOOG4-RSA-SHA256&X-Goog-Credential=481589474394-compute%40developer.gserviceaccount.com%2F20240404%2Fauto%2Fstorage%2Fgoog4_request&X-Goog-Date=20240404T030354Z&X-Goog-Expires=901&X-Goog-SignedHeaders=host&X-Goog-Signature=9147d099f165b04095e5f7653f7c4bfd02d9620d688a5e48ae50f9ff6219d0e02b26ae1a976545f07e3b60c56947fbdbb3fd36a0e8b5510cbaf0a3ed1e2a2ce51fd25488e3a9e688236f0d99c3e6e8b43eb60f35886471b633b9e4602f54585b19f72b1072ab50e267a89d7087cdf96b52ab26f53907302e95eefb5120d6cec04db2f3c14d2f88f70cf4add88489a024e29c8a27bf0225dc90f14fbd2d6b62a0777e6aea8a

In [ ]:
# !wget -O Aquarium_Data.zip https://public.roboflow.com/ds/JD9wKRLpmV?key=hNydIRTP9Y

In [ ]:
import zipfile

with zipfile.ZipFile('/content/Firearms.zip') as target_file:
  target_file.extractall('/content/Firearms/')

<h1>YAML 파일 확인하기

In [ ]:
!cat /content/Fireamrs/data.yaml

NotImplementedError: A UTF-8 locale is required. Got ANSI_X3.4-1968

<h1> YAML_DATA COLAB 환경에 맞게 수정하기

In [ ]:
!pip install PyYAML

In [ ]:
import yaml

data = {'train' : '/content/Firearms/train/images',
        'val' : '/content/Firearms/valid/images',
        'test' : '/content/Firearms/test/images',
        'names' : ['pistol','rifle'],
        'nc' : 2}

with open('/content/Firearms/data.yaml', 'w') as f:
  yaml.dump(data, f)
with open('/content/Firearms/data.yaml', 'r') as f:
  Firearms_yaml = yaml.safe_load(f)
  display(Firearms_yaml)

{'names': ['pistol', 'rifle'],
 'nc': 2,
 'test': '/content/Firearms/test/images',
 'train': '/content/Firearms/train/images',
 'val': '/content/Firearms/valid/images'}

<h1> YOLOv8 설치 및 데이터 로드

In [ ]:
!pip install ultralytics

In [ ]:
import ultralytics
ultralytics.checks()

Ultralytics YOLOv8.1.42 🚀 Python-3.10.12 torch-2.2.1+cu121 CUDA:0 (Tesla T4, 15102MiB)
Setup complete ✅ (8 CPUs, 51.0 GB RAM, 29.4/201.2 GB disk)


In [ ]:
from ultralytics import YOLO
model = YOLO('yolov8x.pt')

In [ ]:
print(type(model.names), len(model.names))
print(model.names)

<class 'dict'> 80
{0: 'person', 1: 'bicycle', 2: 'car', 3: 'motorcycle', 4: 'airplane', 5: 'bus', 6: 'train', 7: 'truck', 8: 'boat', 9: 'traffic light', 10: 'fire hydrant', 11: 'stop sign', 12: 'parking meter', 13: 'bench', 14: 'bird', 15: 'cat', 16: 'dog', 17: 'horse', 18: 'sheep', 19: 'cow', 20: 'elephant', 21: 'bear', 22: 'zebra', 23: 'giraffe', 24: 'backpack', 25: 'umbrella', 26: 'handbag', 27: 'tie', 28: 'suitcase', 29: 'frisbee', 30: 'skis', 31: 'snowboard', 32: 'sports ball', 33: 'kite', 34: 'baseball bat', 35: 'baseball glove', 36: 'skateboard', 37: 'surfboard', 38: 'tennis racket', 39: 'bottle', 40: 'wine glass', 41: 'cup', 42: 'fork', 43: 'knife', 44: 'spoon', 45: 'bowl', 46: 'banana', 47: 'apple', 48: 'sandwich', 49: 'orange', 50: 'broccoli', 51: 'carrot', 52: 'hot dog', 53: 'pizza', 54: 'donut', 55: 'cake', 56: 'chair', 57: 'couch', 58: 'potted plant', 59: 'bed', 60: 'dining table', 61: 'toilet', 62: 'tv', 63: 'laptop', 64: 'mouse', 65: 'remote', 66: 'keyboard', 67: 'cell p

<h1> YOLOv8 커스텀 데이터 학습하기

In [ ]:
 model.train(data='/content/Firearms/data.yaml', epochs = 100, patience=30, batch=30, imgsz=416)

Ultralytics YOLOv8.1.42 🚀 Python-3.10.12 torch-2.2.1+cu121 CUDA:0 (Tesla T4, 15102MiB)
engine/trainer: task=detect, mode=train, model=yolov8x.pt, data=/content/Firearms/data.yaml, epochs=100, time=None, patience=30, batch=30, imgsz=416, save=True, save_period=-1, cache=False, device=None, workers=8, project=None, name=train2, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, freeze=None, multi_scale=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, embed=None, show=False, save_frames=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, sho

train: Scanning /content/Firearms/train/labels.cache... 3077 images, 0 backgrounds, 0 corrupt: 100%|██████████| 3077/3077 [00:00<?, ?it/s]


albumentations: Blur(p=0.01, blur_limit=(3, 7)), MedianBlur(p=0.01, blur_limit=(3, 7)), ToGray(p=0.01), CLAHE(p=0.01, clip_limit=(1, 4.0), tile_grid_size=(8, 8))


val: Scanning /content/Firearms/valid/labels.cache... 294 images, 1 backgrounds, 0 corrupt: 100%|██████████| 294/294 [00:00<?, ?it/s]


Plotting labels to runs/detect/train2/labels.jpg... 
optimizer: 'optimizer=auto' found, ignoring 'lr0=0.01' and 'momentum=0.937' and determining best 'optimizer', 'lr0' and 'momentum' automatically... 
optimizer: AdamW(lr=0.001667, momentum=0.9) with parameter groups 97 weight(decay=0.0), 104 weight(decay=0.00046875), 103 bias(decay=0.0)
TensorBoard: model graph visualization added ✅
Image sizes 416 train, 416 val
Using 8 dataloader workers
Logging results to runs/detect/train2
Starting training for 100 epochs...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      1/100      11.8G      1.437      1.807      1.707         46        416: 100%|██████████| 103/103 [01:39<00:00,  1.03it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.20it/s]

                   all        294        430    0.00474      0.205    0.00231   0.000777



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      2/100      11.5G      1.635      1.888      1.845         55        416: 100%|██████████| 103/103 [01:41<00:00,  1.02it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430      0.552     0.0496     0.0164    0.00652



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      3/100      11.5G       1.58      1.799      1.792         60        416: 100%|██████████| 103/103 [01:43<00:00,  1.00s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.16it/s]

                   all        294        430      0.582      0.126     0.0473     0.0188



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      4/100      11.5G      1.502      1.712      1.719         53        416: 100%|██████████| 103/103 [01:43<00:00,  1.00s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.17it/s]

                   all        294        430      0.749      0.287      0.239      0.128



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      5/100      11.5G       1.42      1.616      1.667         62        416: 100%|██████████| 103/103 [01:43<00:00,  1.00s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.17it/s]

                   all        294        430      0.869        0.3      0.351      0.226



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      6/100      11.5G      1.352      1.532      1.613         55        416: 100%|██████████| 103/103 [01:43<00:00,  1.00s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.17it/s]

                   all        294        430      0.877       0.29      0.327      0.203



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      7/100      11.5G       1.29      1.418      1.556         44        416: 100%|██████████| 103/103 [01:43<00:00,  1.01s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.17it/s]

                   all        294        430      0.794      0.217      0.235      0.134



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      8/100      11.5G      1.259      1.371       1.54         64        416: 100%|██████████| 103/103 [01:43<00:00,  1.01s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.16it/s]

                   all        294        430      0.923      0.329      0.413      0.287



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      9/100      11.5G      1.242      1.352      1.536         45        416: 100%|██████████| 103/103 [01:43<00:00,  1.00s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.16it/s]

                   all        294        430       0.92      0.345      0.425      0.309



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     10/100      11.5G      1.204      1.296      1.513         59        416: 100%|██████████| 103/103 [01:43<00:00,  1.00s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.17it/s]

                   all        294        430      0.951       0.31      0.443      0.329



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     11/100      11.5G       1.15      1.224      1.475         49        416: 100%|██████████| 103/103 [01:43<00:00,  1.00s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.17it/s]

                   all        294        430       0.83      0.451      0.501      0.364



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     12/100      11.5G      1.132      1.198       1.46         46        416: 100%|██████████| 103/103 [01:43<00:00,  1.00s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.17it/s]

                   all        294        430      0.937      0.368      0.427      0.301



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     13/100      11.5G      1.144      1.199      1.466         60        416: 100%|██████████| 103/103 [01:43<00:00,  1.00s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.17it/s]

                   all        294        430      0.935      0.356      0.512      0.359



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     14/100      11.5G      1.116      1.164      1.446         53        416: 100%|██████████| 103/103 [01:43<00:00,  1.00s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430      0.611      0.448      0.463      0.327



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     15/100      11.5G      1.101      1.129      1.433         56        416: 100%|██████████| 103/103 [01:43<00:00,  1.00s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430      0.868      0.531      0.553      0.423



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     16/100      11.5G      1.097      1.115      1.434         50        416: 100%|██████████| 103/103 [01:43<00:00,  1.00s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430       0.58      0.545      0.517      0.382



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     17/100      11.5G      1.071      1.081      1.416         50        416: 100%|██████████| 103/103 [01:43<00:00,  1.00s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430      0.781      0.475      0.552      0.395



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     18/100      11.5G      1.067      1.074      1.413         40        416: 100%|██████████| 103/103 [01:43<00:00,  1.00s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.17it/s]

                   all        294        430      0.889      0.537      0.562      0.408



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     19/100      11.5G      1.056      1.036      1.403         64        416: 100%|██████████| 103/103 [01:43<00:00,  1.00s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430      0.723      0.547      0.554      0.416



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     20/100      11.5G      1.029      1.037      1.382         65        416: 100%|██████████| 103/103 [01:43<00:00,  1.00s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.17it/s]

                   all        294        430      0.746      0.547      0.591      0.428



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     21/100      11.5G      1.028      1.016      1.385         51        416: 100%|██████████| 103/103 [01:43<00:00,  1.01s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430      0.915      0.501      0.564      0.423



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     22/100      11.5G      1.013     0.9965      1.378         59        416: 100%|██████████| 103/103 [01:43<00:00,  1.00s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430      0.608      0.558      0.523      0.394



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     23/100      11.5G      1.003     0.9838      1.366         55        416: 100%|██████████| 103/103 [01:43<00:00,  1.00s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.17it/s]

                   all        294        430      0.719      0.514      0.514      0.381



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     24/100      11.5G     0.9901     0.9533      1.351         63        416: 100%|██████████| 103/103 [01:43<00:00,  1.01s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430      0.622      0.533      0.525      0.401



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     25/100      11.5G     0.9844     0.9509      1.359         36        416: 100%|██████████| 103/103 [01:43<00:00,  1.00s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430       0.76      0.524      0.578       0.44



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     26/100      11.5G     0.9805     0.9422      1.345         45        416: 100%|██████████| 103/103 [01:43<00:00,  1.01s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430       0.78       0.51      0.551       0.42



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     27/100      11.5G     0.9743     0.9316      1.347         71        416: 100%|██████████| 103/103 [01:43<00:00,  1.00s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430      0.779      0.508      0.576       0.43



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     28/100      11.5G     0.9543     0.9141      1.334         49        416: 100%|██████████| 103/103 [01:43<00:00,  1.00s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430      0.882      0.536      0.626      0.484



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     29/100      11.5G     0.9483     0.9066      1.326         69        416: 100%|██████████| 103/103 [01:43<00:00,  1.00s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430       0.89      0.517      0.606      0.462



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     30/100      11.5G     0.9372     0.8935      1.315         46        416: 100%|██████████| 103/103 [01:43<00:00,  1.00s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430      0.763      0.535      0.554       0.43



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     31/100      11.5G     0.9362     0.8872      1.323         47        416: 100%|██████████| 103/103 [01:43<00:00,  1.01s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.17it/s]

                   all        294        430      0.697      0.565      0.571      0.431



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     32/100      11.5G     0.9177     0.8593      1.304         60        416: 100%|██████████| 103/103 [01:43<00:00,  1.00s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.17it/s]

                   all        294        430      0.719      0.535      0.584      0.436



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     33/100      11.5G     0.9058     0.8501      1.292         43        416: 100%|██████████| 103/103 [01:43<00:00,  1.00s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430      0.913      0.513      0.616      0.465



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     34/100      11.5G      0.893     0.8395      1.293         49        416: 100%|██████████| 103/103 [01:43<00:00,  1.00s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430       0.94      0.533      0.595      0.471



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     35/100      11.5G     0.8923     0.8231      1.291         55        416: 100%|██████████| 103/103 [01:43<00:00,  1.00s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.17it/s]

                   all        294        430      0.677      0.641       0.66      0.509



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     36/100      11.5G     0.9028      0.835      1.297         57        416: 100%|██████████| 103/103 [01:43<00:00,  1.00s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430      0.975      0.596       0.66      0.525



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     37/100      11.5G     0.8856       0.81      1.287         49        416: 100%|██████████| 103/103 [01:43<00:00,  1.01s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430      0.762      0.556      0.573      0.445



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     38/100      11.5G     0.8892     0.8134       1.28         45        416: 100%|██████████| 103/103 [01:43<00:00,  1.00s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430      0.886      0.581      0.668      0.511



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     39/100      11.5G     0.8713     0.7926      1.273         47        416: 100%|██████████| 103/103 [01:43<00:00,  1.01s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.17it/s]

                   all        294        430      0.943      0.602       0.67      0.522



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     40/100      11.5G     0.8671     0.8004      1.275         76        416: 100%|██████████| 103/103 [01:43<00:00,  1.01s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430      0.943      0.606      0.665      0.534



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     41/100      11.5G      0.877     0.7899      1.269         52        416: 100%|██████████| 103/103 [01:43<00:00,  1.01s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430      0.737      0.532      0.575      0.439



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     42/100      11.5G     0.8435     0.7548      1.251         58        416: 100%|██████████| 103/103 [01:43<00:00,  1.01s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430      0.733      0.611      0.689      0.535



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     43/100      11.5G     0.8509     0.7591      1.259         62        416: 100%|██████████| 103/103 [01:43<00:00,  1.01s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.17it/s]

                   all        294        430      0.838      0.603      0.691      0.548



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     44/100      11.5G     0.8491     0.7633      1.253         54        416: 100%|██████████| 103/103 [01:43<00:00,  1.00s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.17it/s]

                   all        294        430      0.765       0.54      0.613      0.476



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     45/100      11.5G      0.828     0.7423      1.248         54        416: 100%|██████████| 103/103 [01:43<00:00,  1.01s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430      0.882       0.53       0.64       0.52



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     46/100      11.5G     0.8293     0.7344      1.248         68        416: 100%|██████████| 103/103 [01:43<00:00,  1.01s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.17it/s]

                   all        294        430      0.815      0.602      0.665      0.533



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     47/100      11.5G     0.8104     0.7238       1.24         50        416: 100%|██████████| 103/103 [01:43<00:00,  1.01s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.17it/s]

                   all        294        430      0.816      0.609       0.68      0.526



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     48/100      11.5G     0.8195     0.7207      1.231         46        416: 100%|██████████| 103/103 [01:43<00:00,  1.00s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430      0.802       0.63      0.649      0.509



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     49/100      11.5G     0.7995     0.7145       1.23         54        416: 100%|██████████| 103/103 [01:43<00:00,  1.00s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430      0.958      0.607      0.732      0.578



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     50/100      11.5G     0.8053     0.7166      1.231         56        416: 100%|██████████| 103/103 [01:43<00:00,  1.00s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.16it/s]

                   all        294        430      0.864      0.606      0.692      0.534



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     51/100      11.5G     0.8038      0.706      1.229         56        416: 100%|██████████| 103/103 [01:43<00:00,  1.01s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430      0.849      0.707      0.681      0.525



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     52/100      11.5G     0.7906      0.697      1.224         48        416: 100%|██████████| 103/103 [01:43<00:00,  1.00s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430      0.823       0.61      0.649        0.5



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     53/100      11.5G     0.7986     0.6936       1.23         55        416: 100%|██████████| 103/103 [01:43<00:00,  1.01s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.17it/s]

                   all        294        430      0.936      0.601      0.667      0.534



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     54/100      11.5G     0.7785     0.6765      1.213         70        416: 100%|██████████| 103/103 [01:43<00:00,  1.01s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430      0.819      0.611      0.668      0.544



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     55/100      11.5G     0.7798     0.6748      1.211         56        416: 100%|██████████| 103/103 [01:43<00:00,  1.01s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.17it/s]

                   all        294        430      0.646      0.718      0.645      0.523



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     56/100      11.5G     0.7785      0.682      1.216         55        416: 100%|██████████| 103/103 [01:43<00:00,  1.01s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430      0.803      0.626      0.669       0.53



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     57/100      11.5G     0.7753     0.6634      1.207         66        416: 100%|██████████| 103/103 [01:43<00:00,  1.01s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.19it/s]

                   all        294        430      0.932      0.643      0.664      0.525



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     58/100      11.5G     0.7796      0.661       1.21         62        416: 100%|██████████| 103/103 [01:43<00:00,  1.01s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430      0.952      0.705      0.736      0.598



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     59/100      11.5G      0.769     0.6453      1.199         46        416: 100%|██████████| 103/103 [01:43<00:00,  1.01s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.19it/s]

                   all        294        430      0.954      0.608      0.679      0.543



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     60/100      11.5G     0.7436     0.6207      1.186         60        416: 100%|██████████| 103/103 [01:43<00:00,  1.01s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430      0.763      0.542      0.616      0.498



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     61/100      11.5G     0.7385      0.638      1.185         78        416: 100%|██████████| 103/103 [01:43<00:00,  1.01s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430      0.857      0.696      0.721      0.593



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     62/100      11.5G     0.7455     0.6356      1.187         53        416: 100%|██████████| 103/103 [01:43<00:00,  1.01s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430      0.744      0.715       0.71      0.577



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     63/100      11.5G     0.7325     0.6188      1.174         51        416: 100%|██████████| 103/103 [01:43<00:00,  1.01s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430      0.943      0.548      0.642      0.524



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     64/100      11.5G     0.7337     0.6128      1.181         49        416: 100%|██████████| 103/103 [01:43<00:00,  1.01s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430      0.846      0.625      0.647      0.532



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     65/100      11.5G     0.7251     0.6075      1.177         54        416: 100%|██████████| 103/103 [01:43<00:00,  1.00s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430      0.821      0.618      0.633       0.51



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     66/100      11.5G     0.7174     0.6107      1.168         51        416: 100%|██████████| 103/103 [01:43<00:00,  1.00s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.19it/s]

                   all        294        430      0.807      0.615      0.647       0.53



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     67/100      11.5G     0.7102     0.6074      1.171         50        416: 100%|██████████| 103/103 [01:43<00:00,  1.01s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430      0.932      0.607      0.701      0.572



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     68/100      11.5G     0.7047     0.5898      1.168         57        416: 100%|██████████| 103/103 [01:43<00:00,  1.01s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430      0.975       0.69      0.736      0.605



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     69/100      11.5G     0.7122     0.5904      1.167         59        416: 100%|██████████| 103/103 [01:43<00:00,  1.00s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.19it/s]

                   all        294        430      0.851      0.689      0.702      0.585



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     70/100      11.5G     0.6856     0.5684      1.153         59        416: 100%|██████████| 103/103 [01:43<00:00,  1.01s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430       0.96      0.627       0.69      0.549



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     71/100      11.5G     0.6924     0.5758      1.159         50        416: 100%|██████████| 103/103 [01:43<00:00,  1.01s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430      0.956      0.594      0.677      0.555



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     72/100      11.5G     0.6996     0.5694       1.16         68        416: 100%|██████████| 103/103 [01:43<00:00,  1.00s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430      0.912      0.615      0.706      0.569



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     73/100      11.5G     0.6862     0.5708      1.157         54        416: 100%|██████████| 103/103 [01:43<00:00,  1.00s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430      0.925      0.614      0.683      0.554



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     74/100      11.5G     0.6874     0.5642      1.148         54        416: 100%|██████████| 103/103 [01:43<00:00,  1.01s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430      0.769      0.694       0.71      0.594



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     75/100      11.5G     0.6731     0.5494      1.141         49        416: 100%|██████████| 103/103 [01:44<00:00,  1.01s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.19it/s]

                   all        294        430      0.806      0.646      0.646      0.536



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     76/100      11.5G     0.6658     0.5452      1.144         53        416: 100%|██████████| 103/103 [01:43<00:00,  1.00s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430      0.834      0.637      0.654      0.533



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     77/100      11.5G     0.6684     0.5478      1.143         62        416: 100%|██████████| 103/103 [01:43<00:00,  1.01s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430      0.836      0.629       0.68      0.561



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     78/100      11.5G     0.6585     0.5372      1.132         65        416: 100%|██████████| 103/103 [01:43<00:00,  1.01s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430      0.723      0.709      0.708      0.576



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     79/100      11.5G     0.6597     0.5292      1.134         55        416: 100%|██████████| 103/103 [01:43<00:00,  1.01s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.17it/s]

                   all        294        430      0.914      0.624      0.694      0.555



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     80/100      11.5G      0.652     0.5281      1.138         61        416: 100%|██████████| 103/103 [01:43<00:00,  1.00s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.19it/s]

                   all        294        430      0.931      0.624      0.676      0.545



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     81/100      11.5G     0.6499      0.521      1.129         57        416: 100%|██████████| 103/103 [01:44<00:00,  1.01s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430      0.973      0.613      0.693      0.567



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     82/100      11.5G     0.6301     0.5127      1.118         61        416: 100%|██████████| 103/103 [01:43<00:00,  1.01s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430      0.839      0.712      0.716      0.567



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     83/100      11.5G      0.635     0.5182      1.122         49        416: 100%|██████████| 103/103 [01:43<00:00,  1.01s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.17it/s]

                   all        294        430      0.954      0.721      0.743       0.62



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     84/100      11.5G     0.6325     0.5106       1.12         42        416: 100%|██████████| 103/103 [01:43<00:00,  1.01s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430      0.954      0.627      0.704      0.569



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     85/100      11.5G      0.637     0.5071      1.121         48        416: 100%|██████████| 103/103 [01:43<00:00,  1.01s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430      0.962      0.623      0.703      0.577



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     86/100      11.5G     0.6105     0.4972       1.11         50        416: 100%|██████████| 103/103 [01:43<00:00,  1.01s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430      0.955      0.617      0.694      0.556



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     87/100      11.5G     0.6195     0.5028      1.114         51        416: 100%|██████████| 103/103 [01:43<00:00,  1.01s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430      0.869      0.696       0.72      0.594



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     88/100      11.5G     0.6143      0.494      1.112         61        416: 100%|██████████| 103/103 [01:43<00:00,  1.01s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.17it/s]

                   all        294        430      0.967      0.696       0.74      0.603



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     89/100      11.5G     0.6172     0.4947      1.111         55        416: 100%|██████████| 103/103 [01:43<00:00,  1.01s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430      0.824      0.624      0.669      0.554



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     90/100      11.5G     0.6106     0.4898      1.112         55        416: 100%|██████████| 103/103 [01:43<00:00,  1.01s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430      0.851      0.701      0.736      0.602


Closing dataloader mosaic
albumentations: Blur(p=0.01, blur_limit=(3, 7)), MedianBlur(p=0.01, blur_limit=(3, 7)), ToGray(p=0.01), CLAHE(p=0.01, clip_limit=(1, 4.0), tile_grid_size=(8, 8))

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     91/100      11.5G     0.4516     0.3089     0.9784         27        416: 100%|██████████| 103/103 [01:44<00:00,  1.02s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.17it/s]

                   all        294        430      0.966      0.707      0.763      0.615



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     92/100      11.5G     0.4311     0.2882     0.9635         23        416: 100%|██████████| 103/103 [01:43<00:00,  1.01s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430      0.965      0.707      0.743      0.606



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     93/100      11.5G     0.4232     0.2804     0.9574         25        416: 100%|██████████| 103/103 [01:43<00:00,  1.00s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.17it/s]

                   all        294        430      0.862      0.714      0.712      0.591



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     94/100      11.5G     0.4134     0.2755     0.9556         23        416: 100%|██████████| 103/103 [01:43<00:00,  1.00s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430      0.862      0.721      0.727      0.604



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     95/100      11.5G     0.4051     0.2683     0.9492         26        416: 100%|██████████| 103/103 [01:43<00:00,  1.01s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430      0.968        0.7      0.745      0.617



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     96/100      11.5G      0.402     0.2587     0.9497         32        416: 100%|██████████| 103/103 [01:43<00:00,  1.00s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.17it/s]

                   all        294        430      0.954      0.702      0.739      0.616



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     97/100      11.5G      0.398     0.2588      0.948         33        416: 100%|██████████| 103/103 [01:43<00:00,  1.01s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430      0.962      0.698       0.74      0.607



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     98/100      11.5G     0.3929     0.2542     0.9433         21        416: 100%|██████████| 103/103 [01:43<00:00,  1.00s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.17it/s]

                   all        294        430      0.958      0.702      0.741      0.609



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     99/100      11.5G     0.3858     0.2508     0.9429         23        416: 100%|██████████| 103/103 [01:43<00:00,  1.00s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.18it/s]

                   all        294        430      0.955      0.704      0.741      0.617



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    100/100      11.5G     0.3842     0.2469     0.9395         33        416: 100%|██████████| 103/103 [01:43<00:00,  1.00s/it]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:04<00:00,  1.17it/s]

                   all        294        430      0.957      0.704       0.74      0.607



100 epochs completed in 3.047 hours.
Optimizer stripped from runs/detect/train2/weights/last.pt, 136.7MB
Optimizer stripped from runs/detect/train2/weights/best.pt, 136.7MB

Validating runs/detect/train2/weights/best.pt...
Ultralytics YOLOv8.1.42 🚀 Python-3.10.12 torch-2.2.1+cu121 CUDA:0 (Tesla T4, 15102MiB)
Model summary (fused): 268 layers, 68125494 parameters, 0 gradients, 257.4 GFLOPs


                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 5/5 [00:06<00:00,  1.23s/it]


                   all        294        430      0.954      0.721      0.743       0.62
                pistol        294          7          1       0.57      0.593      0.515
                 rifle        294        423      0.909      0.872      0.894      0.725
Speed: 0.1ms preprocess, 11.4ms inference, 0.0ms loss, 1.0ms postprocess per image
Results saved to runs/detect/train2


ultralytics.utils.metrics.DetMetrics object with attributes:

ap_class_index: array([0, 1])
box: ultralytics.utils.metrics.Metric object
confusion_matrix: <ultralytics.utils.metrics.ConfusionMatrix object at 0x797f4cab16c0>
curves: ['Precision-Recall(B)', 'F1-Confidence(B)', 'Precision-Confidence(B)', 'Recall-Confidence(B)']
curves_results: [[array([          0,    0.001001,    0.002002,    0.003003,    0.004004,    0.005005,    0.006006,    0.007007,    0.008008,    0.009009,     0.01001,    0.011011,    0.012012,    0.013013,    0.014014,    0.015015,    0.016016,    0.017017,    0.018018,    0.019019,     0.02002,    0.021021,    0.022022,    0.023023,
          0.024024,    0.025025,    0.026026,    0.027027,    0.028028,    0.029029,     0.03003,    0.031031,    0.032032,    0.033033,    0.034034,    0.035035,    0.036036,    0.037037,    0.038038,    0.039039,     0.04004,    0.041041,    0.042042,    0.043043,    0.044044,    0.045045,    0.046046,    0.047047,
          0.04804

<h1>모델 분석결과

 <h2>results_dict

:{'metrics/precision(B)': 0.8982572404516258, 'metrics/recall(B)': 0.787986455044358, 'metrics/mAP50(B)': 0.8617371147808082, 'metrics/mAP50-95(B)': 0.6651371682656549, 'fitness': 0.6847971629171703}

--------------------------------------------------------------------------------

metrics/precision(B): 클래스 B에 대한 정밀도(Precision)입니다. 정밀도는 모델이
예측한 클래스 B에 속하는 객체 중 실제로 클래스 B에 속하는 객체의 비율을 나타냅니다. 여기서는 0.898로, 모델이 클래스 B로 예측한 객체 중 약 89.8%가 실제로 클래스 B에 속한다는 것을 의미합니다.

metrics/recall(B): 클래스 B에 대한 재현율(Recall)입니다. 재현율은 실제로 클래스 B에 속하는 객체 중 모델이 올바르게 탐지한 객체의 비율을 나타냅니다. 여기서는 0.788으로, 실제 클래스 B에 속하는 객체 중 약 78.8%가 모델에 의해 올바르게 탐지되었다는 것을 의미합니다.

metrics/mAP50(B): 클래스 B에 대한 평균 정밀도(mAP)입니다. mAP50은 모델이 50%의 IoU (Intersection over Union) 임계값을 기준으로 계산된 평균 정밀도를 나타냅니다. 여기서는 0.862로, 모델이 클래스 B를 포함한 모든 클래스에 대해 상대적으로 높은 정밀도를 보인다는 것을 의미합니다.

metrics/mAP50-95(B): 클래스 B에 대한 평균 정밀도(mAP)입니다. mAP50-95는 모델이 50%에서 95%까지의 다양한 임계값을 사용하여 계산된 평균 정밀도를 나타냅니다. 여기서는 0.665로, 모델이 클래스 B를 포함한 모든 클래스에 대해 일정 범위 내에서 일관된 성능을 보인다는 것을 의미합니다.

fitness: 모델의 전체적인 성능을 나타내는 지표입니다. 일반적으로 여러 지표의 조합으로 계산됩니다. 여기서는 0.685로, 모델의 성능이 상당히 높지는 않지만, 여러 측면에서 괜찮은 성능을 보인다는 것을 의미합니다.


In [ ]:
# import torch

# # 모델 저장
# torch.save(model.state_dict(runs/detect/train), '/content/drive/MyDrive/YOLOv8')

# # 모델 로드
# model.load_state_dict(torch.load('/content/drive/MyDrive/YOLOv8'))
# model.eval()

<h1>Detect/train 데이터들 구글 드라이브에 저장하기!!!!!

In [ ]:
import os
import shutil
from google.colab import drive

# Google 드라이브 마운트
drive.mount('/content/drive')

# 원본 경로
source_path = '/content/runs/detect/train2'

# 목적지 경로
destination_path = '/content/drive/MyDrive/Train_Result_Data/Train4_x'

# 목적지 경로가 없으면 생성
if not os.path.exists(destination_path):
    os.makedirs(destination_path)

# '/content' 경로에 있는 모든 파일과 폴더를 이동
for filename in os.listdir(source_path):
    file_path = os.path.join(source_path, filename)
    if os.path.isfile(file_path) or os.path.isdir(file_path):
        shutil.move(file_path, destination_path)

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


<h1>Valid Set으로 검사하기!

In [ ]:
model_valid = YOLO('/content/drive/MyDrive/Train_Result_Data/Train4_x/weights/best.pt')

In [ ]:
# Validate the model
metrics = model.val()  # no arguments needed, dataset and settings remembered
metrics.box.map    # map50-95
metrics.box.map50  # map50
metrics.box.map75  # map75
metrics.box.maps   # a list contains map50-95 of each category

Ultralytics YOLOv8.1.42 🚀 Python-3.10.12 torch-2.2.1+cu121 CUDA:0 (Tesla T4, 15102MiB)
Model summary (fused): 268 layers, 68125494 parameters, 0 gradients, 257.4 GFLOPs


val: Scanning /content/Firearms/valid/labels.cache... 294 images, 1 backgrounds, 0 corrupt: 100%|██████████| 294/294 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 19/19 [00:10<00:00,  1.89it/s]


                   all        294        430      0.954      0.721      0.743       0.62
                pistol        294          7          1       0.57      0.593      0.515
                 rifle        294        423      0.909      0.873      0.894      0.726
Speed: 0.1ms preprocess, 24.2ms inference, 0.0ms loss, 4.4ms postprocess per image
Results saved to runs/detect/val2


array([    0.51512,     0.72569])

<h1>저장된 best.pt로 test_data Predict 하기!

In [ ]:
model = YOLO('/content/drive/MyDrive/Train_Result_Data/Train4_x/weights/best.pt')

In [ ]:
model.predict(source='/content/Firearms/test/images/', save=True, save_txt=True)


image 1/147 /content/Firearms/test/images/00aded9a9009a16b3e4342803932c072747a28ecb86ee82bf026a200c13cc6c0_jpg.rf.05816de08175e1bed1e62286117f0f46.jpg: 416x416 1 rifle, 51.1ms
image 2/147 /content/Firearms/test/images/03b141ca0876ceeb412e9197548d5a5d627bab942e6613ecb995088ad0b4a27e_jpg.rf.30ced06d1e0766bdbd8d943e0a64335e.jpg: 416x416 1 rifle, 51.3ms
image 3/147 /content/Firearms/test/images/0674c4ec1763822991a4c9c84426bc1b18faccfb7ba4c6d9aed229910408bfad_jpg.rf.9f912f666b82cd8eb028861b131b9af7.jpg: 416x416 2 rifles, 40.7ms
image 4/147 /content/Firearms/test/images/0afd4a8b3eea02c7e6710cb77efc800eeabf2e0c7b7c4da6da4e8f5873476d0e_jpg.rf.13376657f7d34f48c4c5c5e3bc3e344a.jpg: 416x416 1 rifle, 31.8ms
image 5/147 /content/Firearms/test/images/0b6ffde7051a8e67fdc371d49ee11f4c8ee1415bcfd2ce6a383953051037bc4f_jpg.rf.afc0b58a4d8bed4b41501daaf4c20a4e.jpg: 416x416 1 rifle, 31.5ms
image 6/147 /content/Firearms/test/images/0d0034dea636b237a5671ff58e46d0a3d2e0aff0d6b55a17753902358f5d6bd5_jpg.rf.6f82

[ultralytics.engine.results.Results object with attributes:
 
 boxes: ultralytics.engine.results.Boxes object
 keypoints: None
 masks: None
 names: {0: 'pistol', 1: 'rifle'}
 obb: None
 orig_img: array([[[238, 226, 220],
         [233, 221, 215],
         [227, 215, 209],
         ...,
         [169, 186, 195],
         [186, 201, 210],
         [209, 224, 233]],
 
        [[245, 233, 227],
         [237, 225, 219],
         [230, 218, 212],
         ...,
         [169, 186, 195],
         [184, 199, 208],
         [206, 221, 230]],
 
        [[247, 235, 229],
         [235, 223, 217],
         [228, 216, 210],
         ...,
         [172, 189, 198],
         [192, 207, 216],
         [216, 231, 240]],
 
        ...,
 
        [[218, 225, 228],
         [215, 222, 225],
         [215, 222, 225],
         ...,
         [166, 187, 195],
         [169, 190, 198],
         [160, 181, 189]],
 
        [[218, 225, 228],
         [215, 222, 225],
         [215, 222, 225],
         ...,
      

<h1>Valid Set 데이터들 구글 드라이브에 저장하기

In [ ]:
# # Google 드라이브 마운트
# drive.mount('/content/drive')

# 원본 경로
source_path = '/content/runs/detect/val2'

# 목적지 경로
destination_path = '/content/drive/MyDrive/Test_Result_Data/val'

# 목적지 경로가 없으면 생성
if not os.path.exists(destination_path):
    os.makedirs(destination_path)

# '/content' 경로에 있는 모든 파일과 폴더를 이동
for filename in os.listdir(source_path):
    file_path = os.path.join(source_path, filename)
    if os.path.isfile(file_path) or os.path.isdir(file_path):
        shutil.move(file_path, destination_path)

<h1>Predict_test 데이터들 구글 드라이브에 저장하기

In [ ]:
# # Google 드라이브 마운트
# drive.mount('/content/drive')

# 원본 경로
source_path = '/content/runs/detect/predict2'

# 목적지 경로
destination_path = '/content/drive/MyDrive/Test_Result_Data/Test4'

# 목적지 경로가 없으면 생성
if not os.path.exists(destination_path):
    os.makedirs(destination_path)

# '/content' 경로에 있는 모든 파일과 폴더를 이동
for filename in os.listdir(source_path):
    file_path = os.path.join(source_path, filename)
    if os.path.isfile(file_path) or os.path.isdir(file_path):
        shutil.move(file_path, destination_path)

<h1> Drop_Out을 YOLOv8 모델에 적용해보기

In [ ]:
import torch
import torch.nn as nn
from torchvision.models import resnet50

class YOLOv8(nn.Module):
    def __init__(self, num_classes, dropout_prob=0.5):
        super(YOLOv8, self).__init__()
        self.features = resnet50(pretrained=True)
        self.features.fc = nn.Sequential()  # Remove fully connected layer

        # Add dropout after convolutional layers
        self.dropout1 = nn.Dropout2d(p=dropout_prob)
        self.dropout2 = nn.Dropout2d(p=dropout_prob)

        self.conv1 = nn.Conv2d(2048, 1024, kernel_size=3, padding=1)
        self.conv2 = nn.Conv2d(1024, num_classes, kernel_size=1)

    def forward(self, x):
        x = self.features(x)
        x = self.dropout1(x)
        x = self.conv1(x)
        x = self.dropout2(x)
        x = self.conv2(x)
        return x

# 예시로 YOLOv8 모델을 생성합니다. num_classes는 클래스의 개수를 나타냅니다.
model = YOLOv8(num_classes=10)

<h1> Costomed-Detect Model을 이용해서 Track 기능 사용하

<h1> 앙상블을 YOLOv8에 적용해보기

In [ ]:
pass

<h1> 모션 인지 모델이용 하기

In [ ]:
# 찾아 봐야할 모션 인지 모델들....
# google vertex AI
# OpenPose
# AlphaPose
# DensePose

<h1> 구글에서 나오는 cv2_imshow를 사용해서 확인하기

In [ ]:
#pytoch 설치 필요 => detect.py를 사용하기위함이다.
from google.colab.patches import cv2_imshow
import cv2
import os, sys
# !pip install requests

<h1> Argument를 이용해보기(하이퍼파라미터수정을위함)

-하지만 코랩 환경에서는 Argparse를 사용할수 없다. 대신 easydict를 사용해서 구현가능

In [ ]:
# import argparse
# import easydict
from easydict import EasyDict

In [ ]:
# ap = argparse_ArgumentParse()
# ap.add_argumnet('-o', '--output', type=int ,default=50, help='')
# ap.add_argumnet('-b', '--buffer-size', type=int, default =50, help='')
# args = vars(ap.parse_args())

# args = easydict.EasyDict({"output" :  output, "buffer-size" : buffer_size})
args = EasyDict()
args.output = 50 #추가로 필요한 것들은 args.you_need = you_want_data 형식으로 작성